### ISSUE TITLE
No typescript types?

### ISSUE THREAD

**Issue by burtonator** [TimboKZ/react-context-menu-wrapper#1]
I saw your post on reddit about this and looks close to what I'm looking for.

No typescript types though?

**Comment by v1ndic4te** [comment]
All of the relevant types are defined in JSDoc comments for standalone classes and util functions. For React components, the types are defined in `propTypes`. IDEs should be able to pick up these types - PhpStorm parses them just fine.

If you need an explicit `.d.ts` file, I imagine it wouldn't be too hard to generate it based on `src/index.js` using a tool like [tsd-jsdoc](https://www.npmjs.com/package/tsd-jsdoc). If you'll end up doing that, I'll be happy to merge a pull request from you. Not sure when I'll get around to doing that myself.

**Comment by burtonator** [comment]
sure... I banged out a d.ts file real quick... will submit a PR later but posting it here in case I get distracted.  

also need to add:
```
  "types": "./src/index.d.ts",
```

to the package.json file.

```
export class ContextMenuWrapper {
    id: string | null;
    global: boolean;
    onShow: () => void | null;
    onHide: () => void | null;
    hideOnSelfClick: boolean;
    hideOnOutsideClick: boolean;
    hideOnScroll: boolean;
    hideOnWindowResize: boolean;
}


export function prepareContextMenuHandlers(params?: PrepareContextMenuHandlersParams);

export interface PrepareContextMenuHandlersParams {
    readonly id?: string;
    readonly data?: any;
}

export function addContextMenuEventListener(id: string | null,
                                            listener: ContextMenuEventListener);

export function removeContextMenuEventListener(id: string | null, listener: ContextMenuEventListener);

export type ContextMenuEventListener = (eventName: string, data: any | null, publicProps: any) => void


export function hideAllContextMenus();

export function cancelOtherContextMenus();

export enum ContextMenuEvent {
    Show, Hide
}

export function showContextMenu(data: ShowContextMenuData)

export interface ShowContextMenuData {
    readonly id?: string;
    readonly data?: any;
    readonly event?: MouseEvent | ContextMenuEvent;
    readonly x?: any;
    readonly y?: any;
}
```

**Comment by TimboKZ** [comment]
Merged in #5, thanks!
